plugins/inputs/docker/docker.go

```diff
@@ -411,6 +420,11 @@ func (d *Docker) gatherContainer(
 	}
 	daemonOSType := r.OSType
 
+	// use common (printed at `docker ps`) name for container
+	if cname != strings.TrimPrefix(v.Name, "/") && v.Name != "" {
```

Comment:
Don't check if it is already correct, it will be faster just to set it.  Also if you have never seen an empty Name field in docker then I think you should just set the value without any test.